### ISSUE TITLE
new rule: no-listener-leaks

### ISSUE THREAD

**Issue by Harley-Adams** [microsoft/tslint-microsoft-contrib#161]
It’s easy to add an event listener to a DOM element and forget about it.

``` typescript
// leaky
create() {
    this.div.addEventListener("click", () => console.log("hi!"));
}
```

This is unfortunately a common source for memory leaks and/or weird behavior. If the event listener is added multiple times or never removed it will stay there or get continuously added.

``` typescript
// less leaky
create() {
      this.divEvent = this.div.addEventListener("click", () => console.log("hi!"));
}
dispose() {
      this.div.removeEventListener("click", this.divEvent);
}
```

A lint rule that requires the return value of addEventListener be used would help stop this kind of leaky behavior.

**Comment by HamletDRC** [comment]
Good idea. This has some overlap with #37  in concept. 
They should all be rolled into the same rule.

**Comment by JoshuaKGoldberg** [comment]
> requires the return value of addEventListener

I think this makes the most sense as a starting point for the rule. Generally, if we can have a rule that requires the return values of some configurable whitelist of things be used, that can help greatly with this problem.

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated**; therefore, it and tslint-microsoft-contrib are no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint or tslint-microsoft-contrib locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_
